src/lib.rs

```diff
@@ -4249,31 +4182,28 @@ fn autodetect_android_compiler(target: &str, host: &str, gnu: &str, clang: &str)
     // if not, use clang
     if Command::new(&gnu_compiler).output().is_ok() {
         gnu_compiler
-    } else if host.contains("windows") && Command::new(&clang_compiler_cmd).output().is_ok() {
+    } else if cfg!(windows) && Command::new(&clang_compiler_cmd).output().is_ok() {
         clang_compiler_cmd
     } else {
         clang_compiler
     }
 }
 
 // Rust and clang/cc don't agree on how to name the target.
-fn map_darwin_target_from_rust_to_compiler_architecture(target: &str) -> Option<&'static str> {
-    if target.contains("x86_64h") {
-        Some("x86_64h")
-    } else if target.contains("x86_64") {
-        Some("x86_64")
-    } else if target.contains("arm64e") {
-        Some("arm64e")
-    } else if target.contains("aarch64") {
-        Some("arm64")
-    } else if target.contains("i686") {
-        Some("i386")
-    } else if target.contains("powerpc") {
-        Some("ppc")
-    } else if target.contains("powerpc64") {
-        Some("ppc64")
-    } else {
-        None
+fn map_darwin_target_from_rust_to_compiler_architecture(target: &Target) -> &str {
```

Comment:
I think this should be in `target.rs`

I also wonder if rustc provides these info?
It looks like something that could be available in there.

Reply:
If we do generation, then we might be able to extract it from `llvm_target`.

Reply:
That'd be nice, I think cc has too many hardcoded information in it, and it gets hard to maintain or read very quickly.

Reply:
I think it makes sense to punt on this for later, as we're not guaranteed to be able to know `llvm_target` (e.g. custom target JSON and/or unknown targets), and so we'll have to implement some sort of conversion anyhow.

Reply:
Have opened https://github.com/rust-lang/cc-rs/issues/1249 to track it there